### ISSUE TITLE
Diff against default subvolume

### ISSUE THREAD

**Issue by nioncode** [openSUSE/snapper#269]
Is there a way to diff (list of files as given by `status` is enough) an existing snapshot with the _default_ btrfs subvolume (or even better, an arbitrary subvolume)?

Currently, I have to misuse the `rollback` command for that, by doing:
```
snapper rollback
btrfs subvolume set-default 257 /
```
Unfortunately, this creates two unnecessary snapshots and, more importantly, it changes the default subvolume without a way to prevent it. Thus, I have to manually reset the default subvolume again afterwards.

Or, is there a way to create a snapshot of an arbitrary subvolume instead of always using the current one?

**Comment by aschnell** [comment]
No, so far there is no way to diff with the default subvolume or an arbitrary subvolume.

Depending on your setup/distribution the default subvolume is always a subvolume/snapshot managed by snapper. For the latest openSUSE distribution this is the case (YaST installs the system already into a subvolume). So here you can use the btrfs tools to query the default subvolume and from there lookup the corresponding snapper snapshot number and do the compare.

**Comment by nioncode** [comment]
I'm just gonna briefly describe how I currently use snapper on Arch Linux, maybe there is a better way to do this.

BTRFS root:
```
@root (default subvolume) -> gets mounted as /
@home -> gets mounted as /home
@snapshots\root -> get mounted as /.snapshots
```

Now, I create a new snapper snapshot on each boot in order to always have a snapshot that is known to be in a good state.
In case I mess up the system, I boot into the known 'good' snapshot, which is read only. Then, I want to compare this good snapshot with the latest state in my default subvolume.
The issue is, that I don't have a snapshot of the default subvolume.

Using `snapper rollback`, I can create a snapshot of the default subvolume, do all my diffs, and fix the system.
However, I then have to undo the rollback, i.e. manually set the correct default subvolume.

How would you suggest doing this the right way? Can I set things up, so that I always have an up to date snapshot of the default subvolume available? Does it work by creating a read-write snapshot of the current default subvolume and then always booting this?


Anyway, I still think that both, manually creating a snapshot of the default subvolume (or any subvolume), and diffing against an arbitrary subvolume (that does not need to be a snapshot managed by snapper) would be great to have. Are you keen for accepting a PR for this or is this out of scope for snapper?
